src/App.js

```diff
@@ -10,13 +11,17 @@ import NoMatch from './NoMatch';
 import normalize from './styles/normalize'; // eslint-disable-line no-unused-vars
 import base from './styles/base'; // eslint-disable-line no-unused-vars
 
+const history = createBrowserHistory();
+
 class App extends Component {
   componentDidMount() {
-    window.history.replaceState(null, null, '/');
+    // Allow Drupal redirects to determine the initial path.
+    const search = history.location.search.replace('?q=', '');
```

Comment:
This bit ensures that the initial route is the one requested from Drupal. Also: I removed window.history. We should rather use the react router history